files/common/usr/bin/unpack-image

```diff
@@ -113,43 +113,13 @@ fi
 sha256sum -c SHA256SUMS >/dev/null ||
 	die "image is corrupt; checksums don't match"
 
-tar -xzf payload.tar.gz || die "failed to extract payload.tar.gz"
-rm payload.tar.gz || die "failed to remove payload.tar.gz"
+report_progress_inc 50 "Handoff unpack to prepare script."
 
-#
-# We need to be careful when sourcing this file, since it can conflict
-# with (and clobber) functions and/or variables previously defined.
-#
-# shellcheck disable=SC1091
-. version.info || die "sourcing version.info file failed"
-
-[[ -n "$VERSION" ]] || die "VERSION variable is empty"
-[[ -n "$MINIMUM_VERSION" ]] || die "MINIMUM_VERSION variable is empty"
-[[ -n "$MINIMUM_REBOOT_OPTIONAL_VERSION" ]] ||
-	die "MINIMUM_REBOOT_OPTIONAL_VERSION variable is empty"
-
-if $opt_x; then
-	sed -i \
-		"s/^\(MINIMUM_REBOOT_OPTIONAL_VERSION\)=.*$/\1=$VERSION/" \
-		version.info ||
-		die "'sed -i ... version.info' failed"
-
-	# shellcheck disable=SC1091
-	. version.info || die "sourcing version.info file failed"
-fi
+./prepare ${opt_f:+"-f"} ${opt_s:+"-s"} ${opt_x:+"-x"} "$UNPACK_DIR" ||
```

Comment:
How would you feel about moving the `popd` from a few lines below this one, to occur before this call to `prepare`.. and then we change this line to be:
```
"$UNPACK_DIR/prepare" ${opt_f:+"-f"} ${opt_s:+"-s"} ${opt_x:+"-x"} || die ...
```

My reasoning is this:

 1. This will ensure there's no processes with a reference on `$UNPACK_DIR` when `prepare` is called. While this might not be a problem today, I worry that if the process executing `unpack-image` is holding a reference on `$UNPACK_DIR`, it might restrict what `prepare` can do to that directory. I don't have any concrete examples of why this would be required, but I feel like it makes things a little more clean, as far as the assumptions that `prepare` can have about the state of the system.

 2. I don't see any reason as to why we need to pass `$UNPACK_DIR` as an argument to `prepare`, since `prepare` can easily determine this directory for itself via `"${BASH_SOURCE%/*}"` and/or `readlink -f "${BASH_SOURCE%/*}"`.. We even use this approach in other the upgrade scripts today.. So, we can simplify the interface between `unpack-image` and `prepare`, by removing the `$UPACK_DIR` argument.

Reply:
Yes I had planned to do this but seem to have missed it. No reason for popd in unpack-image to happen after prepare script finishes.

Reply:
Have moved the popd up. Regarding the UNPACK_DIR parameter, I would like to be a little conservative and pass it in. There has been some suggestions that the replacement script to prepare in SUV be written in a different language than bash (python is primary suggestion) and I am not sure how reliable getting the directory using something like `os.path.dirname(__file__)` in python is, compared to the bash equivalent. What do you think?

Reply:
I think it's still simple to get it with python:

> As initialized upon program startup, the first item of this list, path[0], is the directory containing the script that was used to invoke the Python interpreter.

See: https://docs.python.org/3/library/sys.html#sys.path

But if you prefer to pass in the directory using an argument, I'm OK with that.